### ISSUE TITLE
User-defined datatype can't be used as GADT index

### ISSUE THREAD

**Issue by buggymcbugfix** [granule-project/granule#94]
```idris
import Bool

data Foo (b : Bool) where
  Bar : Foo b;
  Baz : Foo True
```

gives me

> Unbound variable error: test.gr:4:3:
>   `True` constructor.

**Comment by dorchard** [comment]
This is because we do not promote user-defined data types automatically. We could do this without too much hassle though...

**Comment by buggymcbugfix** [comment]
Implemented by 1ffe7101 (db6dfb3b on master). We only allow monomorphic constructors at the moment, so I'll open a new issue for that.
